src/main/kotlin/no/nav/syfo/client/graphapi/GraphApiClient.kt

```diff
@@ -0,0 +1,73 @@
+package no.nav.syfo.client.graphapi
+
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.httpClientProxy
+import no.nav.syfo.util.bearerHeader
+import org.slf4j.LoggerFactory
+
+class GraphApiClient(
+    private val azureAdClient: AzureAdClient,
+    private val baseUrl: String,
+    private val relevantSyfoRoller: List<AdRolle>
+) {
+    private val httpClient = httpClientProxy()
+
+    suspend fun hasAccess(
+        // TODO: legg inn callId
+        adRolle: AdRolle,
+        token: String,
+    ): Boolean {
+        val groupList = getRoleList(token = token)
+
+        return isRoleInUserGroupList(
+            groupList = groupList.value,
+            adRolle = adRolle,
+        )
+    }
+
+    suspend fun getRoleList(token: String): GraphApiUserGroupsResponse {
+        // TODO: sjekk cachen først
+        // TODO: add callId for exception handling
+        // TODO: add metrics
+        val oboToken = azureAdClient.getOnBehalfOfToken(
+            scopeClientId = baseUrl,
+            token = token,
+        )?.accessToken ?: throw RuntimeException("Failed to request access to Enhet: Failed to get OBO token")
+        val GRAPHAPI_USER_GROUPS_PATH = "/me/memberOf"
+        val FILTER_QUERY = "\$filter="
+        val url = "$baseUrl/v1.0/$GRAPHAPI_USER_GROUPS_PATH?\$count=true&$FILTER_QUERY"
+        val filter = relevantSyfoRoller.map { it.id }.joinToString(separator = " or ") { "id eq '$it'" }
+        val urlWithFilter = "$url$filter"
+
+        return try {
+            val response: GraphApiUserGroupsResponse = httpClient.get(urlWithFilter) {
```

Comment:
Kunne vært bra med en test på at man kaller api-et med forventet url og filter (for å teste filter-logikken gitt noen roller), men det kommer kanskje? 😃

Reply:
Ja, det blir dekket av integrasjonstesten :)